### ISSUE TITLE
Inconsistent package version number separators (.rc2, -rc3) confuse APT

### ISSUE THREAD

**Issue by sfritz** [elastic/logstash#3145]
The debian packages went from using a "." to a "-" to separate the rc suffix.  Versions are sorted lexicographically, so apt thinks rc2 is newer than rc3.  Not a deal breaker, but it requires doing a forced "downgrade".

ex:
1.5.0.rc2-1
1.5.0-rc3-1

**Comment by suyograo** [comment]
@sfritz Thanks for the issue; we will be fixing this in RC4. We had some mix ups regarding versioning formats. Initially we used the format `x.y.z.pre` but we agreed later on a different format across our projects to be `x.y.z-pre` which is based on semver. 

For RC4, we will be going back to the naming format -- `1.5.0.RC4` which will ensure that `apt-get update` will pick that as the latest. 1.5.0 will still be greater than any pre releases.

For RC3 you can as a workaround, you can use `apt-get install logstash=1.5.0-rc3-1`

**Comment by z0rc** [comment]
This isn't going to work.
```
$ dpkg --compare-versions 1.5.0-1 lt 1.5.0.RC4-1 && echo true
true
```
It means that dpkg considers version number `1.5.0` to be less that `1.5.0.RC4`. To archive what you want you have to use `~` (tilde) as a separator e.g. `1.5.0~rc4`.
```
$ dpkg --compare-versions 1.5.0-1 gt 1.5.0~RC4-1 && echo true
true
```
See [Debian Policy](https://www.debian.org/doc/debian-policy/ch-controlfields.html#s-f-Version) for details.

Also while we're at here please consider dropping `-1` (debian_version). It describes version of packaging of current upstream and I haven't seen that you actually update it or work on it out of the release cycle. The less clutter, the better. Also you can reintroduce it later, if you really need it. No debian_version is the same as `-0`.

**Comment by suyograo** [comment]
Fixed with 1.5.0 GA. We had to use epoch to break the tie
